apollo-federation/src/connectors/json_selection/methods/public/get.rs

```diff
@@ -0,0 +1,1212 @@
+            Shape::error(
+                "Method ->get index 5 out of bounds in array of length 2".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn get_shape_should_error_for_array_with_negative_out_of_bounds_index() {
+        assert_eq!(
+            get_test_shape(
+                vec![WithRange::new(LitExpr::Number(Number::from(-5)), None)],
+                Shape::array([Shape::int([]), Shape::string([])], Shape::none(), [])
+            ),
+            Shape::error(
+                "Method ->get index -3 out of bounds in array of length 2".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn get_shape_should_error_for_array_with_out_of_bounds_index_on_empty_array() {
+        assert_eq!(
+            get_test_shape(
+                vec![WithRange::new(LitExpr::Number(Number::from(0)), None)],
+                Shape::array([], Shape::none(), [])
+            ),
+            Shape::error(
+                "Method ->get index 0 out of bounds in array of length 0".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn get_shape_should_error_for_array_with_string_index() {
+        assert_eq!(
+            get_test_shape(
+                vec![WithRange::new(LitExpr::String("invalid".to_string()), None)],
+                Shape::array([Shape::int([])], Shape::none(), [])
+            ),
+            Shape::error(
+                "Method ->get must be provided an integer argument when applied to an array"
+                    .to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    // TODO: Debug here
+    fn get_shape_should_return_unknown_or_none_for_array_with_unknown_index() {
+        assert_eq!(
+            get_test_shape(
+                vec![WithRange::new(
+                    LitExpr::Path(PathSelection {
+                        path: PathList::Key(
+                            Key::field("a").into_with_range(),
+                            PathList::Empty.into_with_range(),
+                        )
+                        .into_with_range(),
+                    }),
+                    None
+                )],
+                Shape::array([Shape::int([])], Shape::none(), [])
+            ),
+            Shape::one(
+                [Shape::unknown([get_location()]), Shape::none()],
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn get_shape_should_return_property_shape_for_object_with_valid_string_key() {
+        let mut fields = IndexMap::default();
+        fields.insert("key".to_string(), Shape::int([]));
+        fields.insert("other".to_string(), Shape::string([]));
+
+        assert_eq!(
+            get_test_shape(
+                vec![WithRange::new(LitExpr::String("key".to_string()), None)],
+                Shape::object(fields.into_iter().collect(), Shape::none(), [])
```

Comment:
```suggestion
                Shape::object(fields), Shape::none(), [])
```

You won't need these if you `use indexmap::IndexMap` at the top of this test module. The trouble is it's trying to use the `apollo_compiler` special version of `IndexMap`.